modules/ocf_desktop/files/stats/update.sh

```diff
@@ -1,15 +1,19 @@
 #!/bin/bash
-# imvirt spews dmesg errors on stderr
-# https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=842226
-[ "$(imvirt 2>/dev/null)" == "Physical" ] || exit 0 # only count hours on desktops
+
+# don't trigger on desktop-like VMs
+[ "$(facter virtual)" == "physical" ] || exit 0
+
+# state can be "active" "cleanup"
+# "active" implies a current logged-in session
+# "cleanup" triggers closing of the session
+# other files to look at: session-cleanup and session-setup in /etc/lightdm
 
 CUR_USER=$(who | awk '$NF == "(:0)" { print $1 }')
-DATA="state=inactive"
 
-if [ -n "$CUR_USER" ]; then
-	DATA="state=active&user=$CUR_USER"
-fi
+[[ -n "$CUR_USER" ]] && STATE="active" || STATE="$1"
```

Comment:
What does the last part of this do? Could you add a comment about what this does to make it a bit clearer?